packages/circuit-ui/components/TierIndicator/TierIndicator.tsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2025, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { forwardRef, type HTMLAttributes } from 'react';
+
+import { clsx } from '../../styles/clsx.js';
+import type { AsPropType } from '../../types/prop-types.js';
+
+import classes from './TierIndicator.module.css';
+
+export interface TierIndicatorProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * The tier name.
+   */
+  tier?: 'Plus';
+  /**
+   * Choose from 3 font sizes. Defaults to `m`.
+   */
+  size?: 's' | 'm' | 'l';
+  /**
+   * Render the text using any HTML element.
+   */
+  as?: AsPropType;
+}
+const sizeMap: Record<string, { width: string; height: string }> = {
+  s: { width: '36', height: '16' },
+  m: { width: '49', height: '24' },
+  l: { width: '63', height: '28' },
+};
+
+export const TierIndicator = forwardRef<HTMLDivElement, TierIndicatorProps>(
+  (
+    { className, as: Element = 'div', tier = 'Plus', size = 's', ...props },
```

Comment:
Based on the prop type description above and to be consistent with other components, the default size should be "m".

```suggestion
    { className, as: Element = 'div', tier = 'Plus', size = 'm', ...props },
```